File: mobile/lib/theme/design_tokens.dart
import 'package:flutter/material.dart';

/// Centralized design tokens — mirrors `src/lib/common/design-tokens.ts` on web.
///
/// Single source of truth for semantic colors used across all screens.
/// Ensures visual consistency between web and mobile platforms.
class DesignTokens {
  DesignTokens._();

  // ─── Primary palette (aligned with web CSS custom properties) ────────
  /// Navy blue — matches web `--primary: 215 50% 25%` ≈ #203A61
  static const Color primary = Color(0xFF203A61);

  /// Accent gold — matches web `--accent: 38 75% 50%`
  static const Color accent = Color(0xFFDFA515);

  // ─── Risk / Severity colors ──────────────────────────────────────────

  static const Color riskCritical = Color(0xFFEF4444);
  static const Color riskCriticalLight = Color(0xFFFEF2F2);

  static const Color riskHigh = Color(0xFFF59E0B);
  static const Color riskHighLight = Color(0xFFFEFCE8);

  static const Color riskMedium = Color(0xFF3B82F6);
  static const Color riskMediumLight = Color(0xFFEFF6FF);

  static const Color riskLow = Color(0xFF10B981);
  static const Color riskLowLight = Color(0xFFECFDF5);

  // ─── Status colors ───────────────────────────────────────────────────

  static const Color statusSuccess = Color(0xFF16A34A);
  static const Color statusWarning = Color(0xFFD97706);
  static const Color statusError = Color(0xFFDC2626);
  static const Color statusInfo = Color(0xFF203A61);

  // ─── Transaction direction ───────────────────────────────────────────

  static const Color credit = riskLow;
  static const Color creditBg = riskLowLight;
  static const Color debit = riskCritical;
  static const Color debitBg = riskCriticalLight;

  // ─── Status → Color mapping ──────────────────────────────────────────

  /// Returns (background, foreground) colors for a status string.
  static (Color bg, Color fg) statusColors(String status) {
    switch (status.toLowerCase()) {
      // Success states
      case 'posted':
      case 'completed':
      case 'active':
      case 'paid':
      case 'approved':
      case 'verified':
        return (riskLowLight, riskLow);

      // Warning states
      case 'pending':
      case 'processing':
      case 'reviewing':
        return (riskHighLight, riskHigh);

      // Info states
      case 'scheduled':
        return (riskMediumLight, riskMedium);

      // Error states
      case 'declined':
      case 'failed':
      case 'rejected':
      case 'overdue':
      case 'delinquent':
        return (riskCriticalLight, riskCritical);

      // Neutral states
      case 'cancelled':
      case 'frozen':
      case 'locked':
      case 'paused':
      case 'closed':
      default:
        return (const Color(0xFFF3F4F6), const Color(0xFF6B7280));
    }
  }

  // ─── Quick action category colors ────────────────────────────────────

  static const Color actionTransfer = riskMedium;
  static const Color actionTransferBg = riskMediumLight;
  static const Color actionBillPay = riskLow;
  static const Color actionBillPayBg = riskLowLight;
  static const Color actionDeposit = primary;
  static const Color actionCards = riskHigh;
  static const Color actionCardsBg = riskHighLight;
  static const Color actionFindAtm = riskCritical;
  static const Color actionFindAtmBg = riskCriticalLight;
}

File: src/lib/common/design-tokens.ts
/**
 * Centralized Design Tokens
 *
 * Single source of truth for semantic color mappings used across all components.
 * Maps status strings and categories to CSS classes that reference the theme's
 * CSS custom properties (defined in index.css), ensuring dark mode, high-contrast,
 * and tenant theming all work correctly.
 *
 * RULE: Never use raw Tailwind color utilities (bg-red-100, text-green-700, etc.)
 *       for status/semantic colors. Always use these tokens instead.
 */

// ---------------------------------------------------------------------------
// Status → Color class mappings (badges, icons, indicators)
// ---------------------------------------------------------------------------

export type StatusSeverity = "success" | "warning" | "error" | "info" | "neutral";

export interface StatusStyle {
  /** Background + text + border classes for badge/pill usage */
  badge: string;
  /** Icon container classes (rounded-full background + icon color) */
  icon: string;
  /** Text-only class for inline text coloring */
  text: string;
  /** Semantic severity for programmatic use */
  severity: StatusSeverity;
}

const success: StatusStyle = {
  badge: "bg-risk-low-light text-risk-low border-risk-low/20",
  icon: "bg-risk-low-light text-risk-low",
  text: "text-status-success",
  severity: "success",
};

const warning: StatusStyle = {
  badge: "bg-risk-high-light text-risk-high border-risk-high/20",
  icon: "bg-risk-high-light text-risk-high",
  text: "text-status-warning",
  severity: "warning",
};

const error: StatusStyle = {
  badge: "bg-risk-critical-light text-risk-critical border-risk-critical/20",
  icon: "bg-risk-critical-light text-risk-critical",
  text: "text-destructive",
  severity: "error",
};

const info: StatusStyle = {
  badge: "bg-risk-medium-light text-risk-medium border-risk-medium/20",
  icon: "bg-risk-medium-light text-risk-medium",
  text: "text-primary",
  severity: "info",
};

const neutral: StatusStyle = {
  badge: "bg-muted text-muted-foreground border-border",
  icon: "bg-muted text-muted-foreground",
  text: "text-muted-foreground",
  severity: "neutral",
};

/**
 * Maps any status string to a semantic style.
 * Covers all statuses used across banking pages: transactions, transfers,
 * bill pay, cards, deposits, accounts, and standing instructions.
 */
const statusMap: Record<string, StatusStyle> = {
  // Success / Positive
  posted: success,
  completed: success,
  active: success,
  paid: success,
  approved: success,
  verified: success,

  // Warning / Pending
  pending: warning,
  processing: warning,
  reviewing: warning,
  scheduled: info,

  // Error / Negative
  declined: error,
  failed: error,
  rejected: error,
  overdue: error,
  delinquent: error,

  // Neutral / Inactive
  cancelled: neutral,
  frozen: neutral,
  locked: neutral,
  paused: neutral,
  closed: neutral,
  draft: neutral,
  archived: neutral,
  expired: neutral,
  queued: neutral,
  invited: neutral,

  // Additional success
  published: success,
  concluded: success,
  activated: success,
  authorized: success,
  sent: success,
  delivered: success,
  read: success,

  // Additional error
  suspended: error,
  revoked: error,

  // Additional info
  running: info,
};

/**
 * Get the semantic style for a status string.
 * Falls back to neutral for unknown statuses.
 */
export function getStatusStyle(status: string): StatusStyle {
  return statusMap[status.toLowerCase()] ?? neutral;
}

// ---------------------------------------------------------------------------
// Transaction direction colors
// ---------------------------------------------------------------------------

export const transactionColors = {
  credit: {
    icon: "bg-risk-low-light text-risk-low",
    text: "text-status-success",
  },
  debit: {
    icon: "bg-risk-critical-light text-risk-critical",
    text: "",
  },
} as const;

// ---------------------------------------------------------------------------
// Quick action / category colors (Dashboard, More menu)
// Uses the theme's semantic palette to avoid arbitrary hardcoded colors.
// ---------------------------------------------------------------------------

export const actionColors = {
  transfer: "bg-risk-medium-light text-risk-medium",
  billPay: "bg-risk-low-light text-risk-low",
  deposit: "bg-primary/10 text-primary",
  cards: "bg-risk-high-light text-risk-high",
  findAtm: "bg-risk-critical-light text-risk-critical",
} as const;

// ---------------------------------------------------------------------------
// Deposit capture state colors
// ---------------------------------------------------------------------------

export const captureColors = {
  captured: {
    border: "border-status-success",
    bg: "bg-risk-low-light",
    icon: "text-status-success",
    text: "text-status-success font-medium",
  },
  empty: {
    border: "border-muted-foreground/25",
    bg: "bg-muted/30",
    icon: "text-muted-foreground",
    text: "text-muted-foreground",
  },
} as const;

// ---------------------------------------------------------------------------
// Elevation / Shadow tokens
// ---------------------------------------------------------------------------

export const elevation = {
  /** Subtle shadow for cards in default state */
  card: "shadow-sm",
  /** Medium shadow for dropdowns, popovers */
  dropdown: "shadow-md",
  /** Elevated shadow for modals, dialogs */
  modal: "shadow-lg",
  /** Top-level shadow for sticky headers */
  header: "shadow-md",
} as const;
